### ISSUE TITLE
External display does not work after disconnecting/reconnecting

### ISSUE THREAD

**Issue by manWithNoName1** [swaywm/sway#3511]
sway version 1.0-beta.2-270-g1a1133dc (Jan 23 2019, branch 'master')

Thinkpad X230, Manjaro Linux, Dell P2415Q monitor

Steps to reproduce:
External display connected and working
Login to sway (from LightDM OR commandline as well with lightdm disabled) .
Sway works with display visible on both screens, LVDS-1 and DP-1
disconnect dp cable of DP-1
Sway is still visible on LVDS-1
reconnect dp cable
no display visible on DP-1
exit sway and relogin. no issues

after setting WLR_DRM_NO_ATOMIC = 1 as suggested by a user on IRC, no issues with disconnecting/reconnecting

Gist of debug log and config
https://gist.github.com/manWithNoName1/2865d0d197252de39bd02be0ab261573

**Comment by manWithNoName1** [comment]
Addendum: went out for a walk after suspending the laptop with 
"sudo systemctl suspend"
On coming back and moving the mouse to unsuspend, the external display didnt work. Disconnecting and reconnecting the dp cable brought back the display

**Comment by emersion** [comment]
Can you run `echo 0xFF | sudo tee /sys/module/drm/parameters/debug`, then trigger the bug, then `echo 0x00 | sudo tee /sys/module/drm/parameters/debug`, and finally upload your `dmesg` log?

**Comment by emersion** [comment]
```
[ 5654.015955] [drm:drm_atomic_check_only [drm]] checking 00000000209d4052
[ 5654.015972] [drm:drm_atomic_helper_check_modeset [drm_kms_helper]] [CRTC:63:pipe C] mode changed
[ 5654.015983] [drm:drm_atomic_helper_check_modeset [drm_kms_helper]] [CRTC:63:pipe C] enable changed
[ 5654.015993] [drm:drm_atomic_helper_check_modeset [drm_kms_helper]] [CRTC:63:pipe C] active changed
[ 5654.016061] [drm:drm_atomic_helper_check_modeset [drm_kms_helper]] Updating routing for [CONNECTOR:79:DP-1]
[ 5654.016073] [drm:drm_atomic_helper_check_modeset [drm_kms_helper]] [CONNECTOR:79:DP-1] using [ENCODER:78:DP B] on [CRTC:63:pipe C]
[ 5654.016082] [drm:drm_atomic_helper_check_modeset [drm_kms_helper]] [CRTC:63:pipe C] needs all connectors, enable: y, active: y
[ 5654.016103] [drm:drm_atomic_add_affected_connectors [drm]] Adding all current connectors for [CRTC:63:pipe C] to 00000000209d4052
[ 5654.016124] [drm:drm_atomic_add_affected_planes [drm]] Adding all current planes for [CRTC:63:pipe C] to 00000000209d4052
[ 5654.016176] [drm:intel_atomic_check [i915]] [CONNECTOR:79:DP-1] checking for sink bpp constrains
[ 5654.016223] [drm:intel_atomic_check [i915]] clamping display bpp (was 36) to EDID reported max of 24
[ 5654.016269] [drm:intel_dp_compute_config [i915]] DP link computation with max lane count 4 max rate 270000 max bpp 24 pixel clock 262750KHz
[ 5654.016315] [drm:intel_dp_compute_config [i915]] DP lane count 4 clock 270000 bpp 24
[ 5654.016357] [drm:intel_dp_compute_config [i915]] DP link rate required 788250 available 1080000
[ 5654.016413] [drm:intel_atomic_check [i915]] checking fdi config on pipe C, lanes 4
[ 5654.016454] [drm:intel_atomic_check [i915]] only 2 lanes on pipe C: required 4 lanes
[ 5654.016499] [drm:intel_atomic_check [i915]] fdi link bw constraint, reducing pipe bpp to 18
[ 5654.016539] [drm:intel_atomic_check [i915]] checking fdi config on pipe C, lanes 3
[ 5654.016583] [drm:intel_atomic_check [i915]] only 2 lanes on pipe C: required 3 lanes
[ 5654.016625] [drm:intel_atomic_check [i915]] CRTC bw constrained, retrying
[ 5654.016667] [drm:intel_dp_compute_config [i915]] DP link computation with max lane count 4 max rate 270000 max bpp 18 pixel clock 262750KHz
[ 5654.016712] [drm:intel_dp_compute_config [i915]] DP lane count 4 clock 162000 bpp 18
[ 5654.016754] [drm:intel_dp_compute_config [i915]] DP link rate required 591188 available 648000
[ 5654.016794] [drm:intel_atomic_check [i915]] checking fdi config on pipe C, lanes 3
[ 5654.016838] [drm:intel_atomic_check [i915]] only 2 lanes on pipe C: required 3 lanes
[ 5654.016879] [drm:intel_atomic_check [i915]] CRTC fixup failed
[ 5654.016924] [drm:intel_dump_pipe_config [i915]] [CRTC:63:pipe C][failed]
[ 5654.016964] [drm:intel_dump_pipe_config [i915]] output_types: DP (0x80)
[ 5654.017009] [drm:intel_dump_pipe_config [i915]] cpu_transcoder: C, pipe bpp: 18, dithering: 0
[ 5654.017051] [drm:intel_dump_pipe_config [i915]] fdi: lanes: 3; gmch_m: 6122518, gmch_n: 8388608, link_m: 510209, link_n: 524288, tu: 64
[ 5654.017096] [drm:intel_dump_pipe_config [i915]] dp m_n: lanes: 4; gmch_m: 7653148, gmch_n: 8388608, link_m: 425174, link_n: 262144, tu: 64
[ 5654.017136] [drm:intel_dump_pipe_config [i915]] audio: 1, infoframes: 0
[ 5654.017179] [drm:intel_dump_pipe_config [i915]] requested mode:
[ 5654.017200] [drm:drm_mode_debug_printmodeline [drm]] Modeline 0:"3840x2160" 30 262750 3840 3888 3920 4000 2160 2163 2168 2191 0x40 0x9
[ 5654.017243] [drm:intel_dump_pipe_config [i915]] adjusted mode:
[ 5654.017264] [drm:drm_mode_debug_printmodeline [drm]] Modeline 0:"3840x2160" 30 262750 3840 3888 3920 4000 2160 2163 2168 2191 0x40 0x9
[ 5654.017310] [drm:intel_dump_pipe_config [i915]] crtc timings: 262750 3840 3888 3920 4000 2160 2163 2168 2191, type: 0x40 flags: 0x9
[ 5654.017351] [drm:intel_dump_pipe_config [i915]] port clock: 162000, pipe src size: 3840x2160, pixel rate 262750
[ 5654.017396] [drm:intel_dump_pipe_config [i915]] pch pfit: pos: 0x00000000, size: 0x00000000, disabled
[ 5654.017436] [drm:intel_dump_pipe_config [i915]] ips: 0, double wide: 0
[ 5654.017477] [drm:intel_dump_pipe_config [i915]] dpll_hw_state: dpll: 0x0, dpll_md: 0x0, fp0: 0x0, fp1: 0x0
[ 5654.017520] [drm:intel_dump_pipe_config [i915]] planes on this crtc
[ 5654.017561] [drm:intel_dump_pipe_config [i915]] [PLANE:52:primary C] disabled, scaler_id = 0
[ 5654.017604] [drm:intel_dump_pipe_config [i915]] [PLANE:55:sprite C] disabled, scaler_id = 0
[ 5654.017645] [drm:intel_dump_pipe_config [i915]] [PLANE:60:cursor C] disabled, scaler_id = 0
[ 5654.017666] [drm:drm_atomic_check_only [drm]] atomic driver check for 00000000209d4052 failed: -22
```

**Comment by emersion** [comment]
Can you report this to i915, if you're still able to reproduce? https://github.com/swaywm/sway/issues/3443#issuecomment-455958401
